client/scripts/components/Room/Room.js

```diff
@@ -141,24 +139,17 @@ class Room extends Component {
       addReaction
     } = this.props
 
-    const {defaultMessage, showWelcome} = this.state
+    const {defaultMessage, hasPosted, dismissedWelcomeModal} = this.state
 
     if (loading) {
       return <div style={styles.spinner}>
         <CircularProgress />
       </div>
     }
 
-    const myNametag = this.getMyNametag()
-
     // If the user is not logged in, return to the homepage
-    if (!me || !room.nametags || !myNametag) {
```

Comment:
We still want to go to JoinRoom if the user is logged in and hasn't clicked "Join" yet. For example: If I'm already logged into Nametag and I click a link to a room off of twitter I should see the "JoinRoom" component, not the "Room" component.